firebase-firestore/src/main/java/com/google/firebase/firestore/core/RelationFilter.java

```diff
@@ -61,8 +62,10 @@ public boolean matches(Document doc) {
       hardAssert(
           refValue instanceof DocumentKey, "Comparing on key, but filter value not a DocumentKey");
       hardAssert(
-          operator != Operator.ARRAY_CONTAINS,
-          "ARRAY_CONTAINS queries don't make sense on document keys.");
+          operator != Operator.ARRAY_CONTAINS
+              && operator != Operator.ARRAY_CONTAINS_ANY
+              && operator != Operator.IN,
+          "'" + operator.toString() + "' queries don't make sense on document keys.");
```

Comment:
IN makes sense on document keys? Or are we explicitly disallowing batch gets via this API?

/cc @wilhuff @mikelehen

Reply:
Oh!  Thanks for flagging...  For some reason I just assumed it wouldn't work.  But I guess you can create indexes on document key just like any field in the document?  If so, then we should probably relax this.

Reply:
Well IN on key basically becomes an inequality, so the same restrictions apply as key equality (which is also treated as an inequality)

Reply:
@rockwotj  Are document-keys allowed for ARRAY_CONTAINS as well, or is it just allowed for IN and ARRAY_CONTAINS_ANY?

Reply:
Just `IN`, no `ARRAY_CONTAINS` or `ARRAY_CONTAINS_ANY` (the key can't be an array).

This is a good point, I should add tests to make sure people aren't doing __key__ ARRAY_CONTAINS "foo" queries.